cmd/root.go

```diff
@@ -205,6 +205,8 @@ func AddCommands(cmd *cobra.Command) {
 		"Ratio of votes for reaching consensus")
 	cmd.PersistentFlags().Uint64Var(&config.TortoiseBeacon.VotesLimit, "tortoise-beacon-votes-limit",
 		config.TortoiseBeacon.VotesLimit, "Maximum allowed number of votes to be sent")
+	cmd.PersistentFlags().Uint32Var(&config.TortoiseBeacon.BeaconSyncNumBlocks, "tortoise-beacon-sync-num-blocks",
```

Comment:
according to Tal during the R&D meeting, he was not sure whether we should set it to 1, 1.5 or 2 clusters of (800) blocks, as 800 is the security param. it's likely we will have to experiment with this number. i.e. either set it to 1X/1.5X/2X or set the number of blocks.

also, for system testing, i am not sure whether 800 gets reduced to a smaller number. so it's also easier to explicitly set it to a small number, disregarding layer_size and num_layers_per_epoch.

i imagine when we get to a stage where we are sure of the network param, we can actually simplify a lot more.

Reply:
to me this number seems a bit magical, I've seen it mentioned in tortoise weakcoin as well. do you know if there is any dependency between layer size and layers per epoch? or how it was invented?

Reply:
i am not familiar with the tortoise weak coin (for self-healing).
800 is our security parameter in the whitepaper. during the R&D discussion, Tal and Iddo always refer 800 as a cluster. if we see 800 of something, and if more than 2/3 of them agree, then our security assumption holds.

if we just sticks to 800, then i can hard code a value in the code that says beaconSyncNumBlocks := 800.
as far as i know, we only have hare committee size set to 800.

in the R&D meetings, Tal's idea was to wait K layers that satisfy 800 or 1600 blocks.
but layers are ticked. let's say the average blocks per layer is 200. then we wait minimum 4 layers.
however, i don't want to wait for 4 full layers if we collected 800 blocks earlier than 4 layers. that's why i decided to use numBlocks instead of layers.

there is no relation to layers per epoch that i know of.